docs/tsp/typeServerProtocol.ts

```diff
@@ -0,0 +1,383 @@
+        // The version of the file, which is used to track changes to the file. It will be incremented each time the file is modified.
+        version: number;
+        // Optional: If the file is part of a chain of files, this can be used to indicate the URI of the
+        // file that is automatically 'imported' by this file. This is useful for cases like Jupyter notebooks
+        chainedFileUri?: string;
+    }
+
+    // Describes a file that's been closed in the language server which the type server needs to be aware of.
+    export interface TextDocumentCloseParams {
+        uri: string;
+    }
+
+    // Parameters for resolving an import
+    export interface ResolveImportParams {
+        // The URI of the source file where the import is referenced.
+        sourceUri: string;
+        // The descriptor of the imported module.
+        moduleDescriptor: ImportedModuleDescriptor;
+        snapshot: number;
+    }
+
+    // Represents settings that can be sent to the type server.
+
+    export type Settings = {
+        [key: string]: any;
+    };
+
+    // Requests and notifications for the type server protocol.
+    export enum Requests {
+        // First request to initialize the type server.
+        Initialize = 'typeServer/initialize',
+        // Request from client to shut down the type server.
+        Shutdown = 'typeServer/shutdown',
+        // Request from client to get the current snapshot of the type server.
+        // A snapshot is a point-in-time representation of the type server's state, including all loaded files and their types.
+        // A type server should change its snapshot whenever any type it might have returned is no longer valid. Meaning types are
+        // only usable for the snapshot they were returned with.
+        GetSnapshot = 'typeServer/getSnapshot',
+        // Request to get diagnostics for a specific file.
+        GetDiagnostics = 'typeServer/getDiagnostics',
+        // Request to get the version of diagnostics for a specific file.
+        GetDiagnosticsVersion = 'typeServer/getDiagnosticsVersion',
+        // Request to get the type information for a specific node.
+        GetType = 'typeServer/getType',
+        // Request to get the union types for a specific type.
+        GetUnionTypes = 'typeServer/getUnionTypes',
+        // Request to find a member of a type.
+        GetTypeMember = 'typeServer/getTypeMember',
+        // Request to get all overloads of a function or method.
+        GetOverloads = 'typeServer/getOverloads',
+        // Request to get the effective meta class of a type.
+        GetEffectiveMetaClass = 'typeServer/getEffectiveMetaClass',
+        // Request to get the type of a declaration.
+        GetTypeOfDeclaration = 'typeServer/getTypeOfDeclaration',
+        // Request to get symbol declaration information for a node.
+        GetSymbolDeclarationInfo = 'typeServer/getSymbolDeclarationInfo',
+        // Request to get the bound magic method for a type.
```

Comment:
I think this should be changed from `GetBoundMagicMethod` to simply `GetBoundMethod`. There's nothing specific to "magic" methods about this call.